pkg/blockchaininfo/nats_publisher.go

```diff
@@ -167,6 +172,64 @@ func (bu *BUpdatesExtensionState) publishUpdates(updates BUpdatesInfo, nc *nats.
 	return nil
 }
 
+// initHistoryJournal with 100 past entries from State when the node starts.
+func (bu *BUpdatesExtensionState) initHistoryJournal(updates BUpdatesInfo) {
+	// TODO implement this.
+}
+
+func (bu *BUpdatesExtensionState) addSentEntriesToHistoryJournal(updates BUpdatesInfo) {
+	height := updates.BlockUpdatesInfo.Height
+	blockID := updates.BlockUpdatesInfo.BlockID
+
+	historyEntry := HistoryEntry{
+		height:  height,
+		blockID: blockID,
+		entries: updates.ContractUpdatesInfo.AllDataEntries,
+	}
+	bu.historyJournal.Push(historyEntry)
+}
+
+func (bu *BUpdatesExtensionState) rollbackHappened(updates BUpdatesInfo, previousState BUpdatesInfo) bool {
+	if updates.BlockUpdatesInfo.Height > previousState.BlockUpdatesInfo.Height {
+		return false
+	}
```

Comment:
The height can be the same, so that's not always true. Especially when node is on a top of blockchain.